### PR TITLE
Issues/1631 suggest tickers parse result

### PR DESCRIPTION
**Pull request by kanishka-azimi**
There will be about 5 small pull requests (back into the suggest tickers main branch). This is the first one.

### CODE REVIEW

File: src/SymbolSearch.hs [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+      if isSimpleSymbol exchangeGroup
+      then sec
+      else error (printf "no suffix found on symbol %s for exchange %s" symbol exchangeGroup)
+    [sec,sfx] ->
+      let ySfx = maybe
+                  (error (printf "unsupported exchange group %s" exchangeGroup))
+                  id
+                  (mediaToYahooSuffix exchangeGroup sfx)
+      in printf "%s.%s" sec ySfx
+    ps -> error (printf "symbol more than two parts %s" symbol)
```
Comment (hritchie) [inline src/SymbolSearch.hs]: ps could be _

File: src/SymbolSearch.hs [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+mediaToYahooSymbol :: String -> String -> String
+mediaToYahooSymbol exchangeGroup symbol =
+  case splitOn ":" symbol of
+    [sec] ->
+      if isSimpleSymbol exchangeGroup
+      then sec
+      else error (printf "no suffix found on symbol %s for exchange %s" symbol exchangeGroup)
+    [sec,sfx] ->
+      let ySfx = maybe
+                  (error (printf "unsupported exchange group %s" exchangeGroup))
```
Comment (hritchie) [inline src/SymbolSearch.hs]: include sfx in error message to give additional context when debugging?

**Comment by kanishka-azimi** [comment]
I forget what else we mentioned turning into a String newtype, but changing the type of exchange group ate up enough time, so I want to stop here on this pull request.

Need to propagate this change to the remaining pull requests.

**Comment by kanishka-azimi** [comment]
I am going to merge this one back into the feature branch. I implemented what we discussed and it should be good enough now.

**Pull request merged**